### PR TITLE
A4A: add support for new pressable enterprise plans in marketplace

### PR DESCRIPTION
**Pull request by travisw**
Related to: https://github.com/Automattic/automattic-for-agencies-dev/issues/1466
P2: pfSQfS-tF-p2

## Proposed Changes

* Add support for new pressable enterprise plans
* Split existing plans and enterprise up into separate tabs

## Why are these changes being made?

* To allow agencies to self-serve purchasing Pressable plans with higher limits

## Testing Instructions

* Sandbox `public-api.wordpress.com`.
* Visit the Pressable marketplace page (`/marketplace/hosting/pressable`).
* You should see two new tabs - "Shared Resource Plans" and "Signature Shared Resource Plans"
* Under "Shared Resource Plans" you should see the existing Pressable plans
* Under "Signature Shared Resource Plans" you should see the new Enterprise Pressable plans ("Signature")
* Make sure the sliders work as expected.
  * If you have an existing plan the slider should start at the next available plan.
  * The slider under "Signature Shared Resource Plans" should default to the far left.
  * In refer mode the sliders should allow choosing any plan (no minimum).
  * The "More" item should appear as the last option under "Signature Shared Resource Plans".
  * Selecting "More" should show the "Contact Us" details just as before.
  * Filtering by installs or visits should continue to work the same.
  * Add one of the new plans to the cart (don't purchase yet). It should show the appropriate "Signature" plan name.
  * Make sure all the displayed plan limits (installs, visits, storage) match what's shown in the spreadsheet in this post (pfSQfS-tF-p2).
* ~~Once D166722-code has shipped, also test purchasing the plans~~
~~:warning: Purchasing plans before that could cause issues since it writes to production tables before the code is live on production. Likely a minor concern but no need to generate errors.~~
  * Make sure purchasing existing plans works the same.
  * Purchase one of the new plans.
  * Once your existing plan is one of the new ones the slider under "Shared Resource Plans" should be all grey so you can't select a lower plan.
  * Under "Signature Shared Resource Plans", the slider should start at the next available plan and not let you select a lower plan, just like before.
  * Make sure when you have any Signature plan or the Business 150 plan that the "Shared Resource Plans" tab is disabled so users can't select a lower or the same plan.
  * Make sure revoking a plan still works ok.

Screenshot 2024-11-22 at 9 27 52 AM
Screenshot 2024-11-22 at 9 29 34 AM
Screenshot 2024-11-22 at 9 31 35 AM
Screenshot 2024-11-22 at 9 35 43 AM
Screenshot 2024-11-22 at 9 36 04 AM
Screenshot 2024-11-22 at 9 41 36 AM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/pressable-overview/lib/get-pressable-plan.ts [hunk 1]
```diff
@@ -110,12 +111,65 @@ const PLAN_DATA: Record< string, PressablePlan > = {
...
+		slug: 'pressable-enterprise-10',
+		install: 500,
+		visits: 10000000,
+		storage: 1000,
+	},
 };
 
 export default function getPressablePlan( slug: string ) {
-	return PLAN_DATA[ slug ];
+	return addPlanCategory( PLAN_DATA[ slug ] );
```
Comment (andrii-lysenko) [inline client/a8c-for-agencies/sections/marketplace/pressable-overview/lib/get-pressable-plan.ts]: Not a strong opinion, but maybe we could just add this to the object itself, to keep things more straight forward.

**Comment by andrii-lysenko** [comment]
While testing, I noticed that when you own enterprise plan, if switching tab to `Standart` it will pre-select 150 plan and you can add it to the cart. We should not allow downgrade and I'm wondering if we should just disable standart tab or purchase button. Other than that I think it looks good, but would confirm with @jeffgolenski 

Screenshot 2024-11-22 at 11 48 49 AM


On a side note, our code around sliders and marketplace is growing, and maybe we could re-visit it in the future to see if it's need some refactoring.

**Comment by travisw** [comment]
> I noticed that when you own enterprise plan, if switching tab to Standart it will pre-select 150 plan and you can add it to the cart

Ah, nice catch. Yeah it does seem like something should get disabled here

> On a side note, our code around sliders and marketplace is growing, and maybe we could re-visit it in the future to see if it's need some refactoring.

Agreed - I think some of the code in these components could be simplified

File: client/a8c-for-agencies/sections/marketplace/pressable-overview/lib/get-pressable-plan.ts (hunk 1 above)
Comment (travisw) [reply client/a8c-for-agencies/sections/marketplace/pressable-overview/lib/get-pressable-plan.ts]: I like that idea, keeps it simpler I suppose

**Comment by travisw** [comment]
> I noticed that when you own enterprise plan, if switching tab to Standart it will pre-select 150 plan and you can add it to the cart

I've changed it so the "Shared Resource Plans" tab becomes disabled when you already own Business 150 or any Signature plan to prevent the user from buying a lower or the same plan.

**Comment by jeffgolenski** [comment]
@travisw Great work. Tested both regular purchases and referrals. All seems to work well on the agency and client sides! Other than what Andrei pointed out, my only recommendation is a small one. Can we shrink the width of the tabs to be `max-width: 600px;` to occupy a little less space?

CleanShot 2024-11-25 at 12 07 04@2x


After that 🚢

**Review by jeffgolenski — CHANGES_REQUESTED** [review]
Adjust the max width of the tabs and we're golden!

**Comment by travisw** [comment]
> Adjust the max width of the tabs and we're golden!

Max width applied

Screenshot 2024-11-25 at 10 25 10 AM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17019592

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @travisw for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17019592).

**Pull request merged**